lib/uuid_mixin.rb

```diff
@@ -1,17 +1,12 @@
 module UuidMixin
   extend ActiveSupport::Concern
   included do
-    before_validation :set_guid, :on => :create if respond_to?(:before_validation)
+    default_value_for(:guid) { SecureRandom.uuid }
   end
 
   private
 
-  def set_guid
```

Comment:
This seems like the only concerning change.  I don't know all of the extracted repos that include this mixin to know if they use this.

Reply:
Although, why would use call `set_guid` if your class responds to `guid=`... 😕

Reply:
Maybe we should keep this method, make it a no-op that returns `self.guid` and deprecate it?

Reply:
I wouldn't expect any of the plugins to call `set_guid`.  It should be included in the base class, not a provider specific subclass

Reply:
> Although, why would use call `set_guid` if your class responds to `guid=`
>
> Maybe we should keep this method, make it a no-op that returns `self.guid` and deprecate it?

The only reason for set_guid was because it was used as a callback form before_validation.  There should not have been any direct callers (i.e. it should have been private).